packages/core/src/storages/key_value_store.ts

```diff
@@ -374,15 +377,28 @@ export class KeyValueStore {
                 Object.assign(cachedValue, value);
             }
         }
-
+    
         // In this case delete the record.
         if (value === null) return this.client.deleteRecord(key);
-
-        value = maybeStringify(value, optionsCopy);
-
+    
+        // Check if the value is a large object before stringifying it
+        let stringifiedValue: string;
+        try {
+            if (isLargeObject(value)) {
+                // Use big-json to stringify large objects
+                stringifiedValue = await bigJsonStringify(value as object);  // Cast to 'object' to work with big-json
+            } else {
+                // Default stringify for small objects, make sure it returns a string
+                stringifiedValue = JSON.stringify(value);  // We can use JSON.stringify here
```

Comment:
Originally, we're using `maybeStringify` here - and we want to keep it. In case the user passes a typed array of bytes, e.g. 
```javascript
new Uint32Array([0xBEEF, 0xCAFE]), `application/octet-stream`
```

They surely don't want to store `{"0":48879,"1":51966}` (or it's binary representation)  in the KVS.